### ISSUE TITLE
Add easy way to dispatch from app projects

### ISSUE THREAD

**Issue by jimbobbennett** [fsprojects/Fabulous#239]
It would be nice if there was an easy way to dispatch messages from the App projects.

For example permissions - these are requested asynchronously in the Android and iOS app projects, and once accepted or rejected a message could then be dispatched to update the model and UI.

**Comment by TimLariviere** [comment]
I can see the need for things like external events (e.g. deep-linking), where you won't be inside the Fabulous loop.

But I don't think you should use dispatch from the App projects to ask for permissions.
Do you have an example where you can't await inside the Core project?

In ElmishContacts, I used the [PermissionsPlugin](https://github.com/jamesmontemagno/PermissionsPlugin) that allows to request asynchronously for a permission. Coupled with `Cmd.ofAsyncMsgOption`, you can react (or not) based on the choice of the user.
https://github.com/TimLariviere/ElmishContacts/blob/541ded8a94479090988c05a9062dc10c80955396/ElmishContacts/Helpers.fs#L41-L52

Even if PermissionsPlugin doesn't provide a permission I need, I tend to use DependencyService and make an interface that encapsulates the whole "ask for permission" flow in a single awaitable method, called in the Core project.

Globally, one strength of Fabulous is to centralize the events inside init\update\view\subscriptions.
And I think exposing dispatch outside can be dangerous. (a previous global dispatch has been marked obsolete).

**Comment by dsyme** [comment]
> And I think exposing dispatch outside can be dangerous. (a previous global dispatch has been marked obsolete).

I think in the context of xamarin there is nothing dangerous about a global dispatch. However it has to have the right type, ie the message type must be that of the overall application.

The previous global dispatch was broken as it had generic type. 

The 'program' could report a dispatch function for example I think.

**Comment by TimLariviere** [comment]
Accessing the dispatch function through the program runner (exposed only by the App class) can be a good solution.

What worries me is that if we give a static global dispatch, it might be misused instead of correctly flowing the dispatch function given in parameter.
It's the same function, but using the one passed in parameter is way easier to know where a dispatch occurs.

**Comment by jimbobbennett** [comment]
I think we need some way to dispatch messages in a clean simple way. I get your point about using the dependency service, but that feels a bit janky, having to create a class/interface just to pass a message around.

In a couple of projects I've used the Xamarin.Forms messenger to pass messages around, and set up a subscription to that which dispatches messages, but it feels a bit heavyweight and after the third app that uses it, feels like it should be baked in.

Permissions probably wasn't the best example, but I've also used it for platform specific stuff like handling callbacks from ai services in a third-party jar that I've bound.

Global dispatch doesn't feel right, and access to a dispatch function on the program on the App was my first thought.
